src/System.Management.Automation/engine/parser/Compiler.cs

```diff
@@ -5500,16 +5502,28 @@ public object VisitArrayExpression(ArrayExpressionAst arrayExpressionAst)
             }
             values = values ?? CaptureAstResults(subExpr, CaptureAstContext.Enumerable);
 
-            if (values.Type.IsArray)
+            if (values.Type == typeof(object[]) || values.Type == typeof(List<object>))
```

Comment:
The clone was added to address a comment raised by @PetSerAl about `$a = 1,2,3; @([object[]]$a)` doesn't return a new array. But if we only consider `if (subExpr is ArrayLiteralAst)` then the clone and the null check are both not necessary anymore.

The `List<object>` case seems like an optimization for `CaptureAstResults`, because a `List<object>` instance will be created to hold the object written to pipe in `CaptureAstResults`.
Are you suggesting to not cover the cases of `values.Type == typeof(object[])` and `values.Type == typeof(List<object>)` here?